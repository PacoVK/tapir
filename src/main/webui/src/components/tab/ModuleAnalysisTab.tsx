import React from "react";
import { Tab, Tabs, Typography } from "@mui/material";
import { ModuleAnalysisTabProps } from "../../types";
import TabPanel from "./TabPanel";
import ModuleInput from "../documentation/ModuleInput";
import ModuleOutput from "../documentation/ModuleOutput";
import ModuleDependencies from "../documentation/ModuleDependencies";
import ReportFindings from "../securityReportVisualization/ReportFindings";
import ModuleResources from "../documentation/ModuleResources";

const ModuleAnalysisTab = (props: ModuleAnalysisTabProps) => {
  const { version, reports, documentation } = props;
  const [tab, setTab] = React.useState(0);

  const handleTabChange = (event: React.SyntheticEvent, newValue: number) => {
    setTab(newValue);
  };

  return documentation ? (
    <>
      <Tabs
        value={tab}
        onChange={handleTabChange}
        aria-label="module detail tabs"
      >
        <Tab label={`Inputs (${documentation.inputs.length})`} />
        <Tab label={`Outputs (${documentation.outputs.length})`} />
        <Tab
          label={`Dependencies (${
            documentation.modules.length + documentation.providers.length
          })`}
        />
        <Tab label={`Resources (${documentation.resources.length})`} />
        <Tab label="Security Report" />
      </Tabs>
      <TabPanel value={tab} index={0}>
        <ModuleInput inputs={documentation.inputs} />
      </TabPanel>
      <TabPanel value={tab} index={1}>
        <ModuleOutput outputs={documentation.outputs} />
      </TabPanel>
      <TabPanel value={tab} index={2}>
        <ModuleDependencies
          modules={documentation.modules}
          providers={documentation.providers}
        />
      </TabPanel>
      <TabPanel value={tab} index={3}>
        <ModuleResources resources={documentation.resources} />
      </TabPanel>
      <TabPanel value={tab} index={4}>
        {reports ? (
          <ReportFindings reports={reports} />
        ) : (
          <Typography>
            No report available for module version: {version}
          </Typography>
        )}
      </TabPanel>
    </>
  ) : null;
};

export default ModuleAnalysisTab;
