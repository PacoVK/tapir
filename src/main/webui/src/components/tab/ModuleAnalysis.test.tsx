import React from "react";
import { render } from "@testing-library/react";
import ModuleAnalysisTab from "./ModuleAnalysisTab";

const fakeResponse = {
  id: "paco-fooo-aws-1.0.2",
  moduleName: "fooo",
  moduleVersion: "1.0.2",
  moduleNamespace: "paco",
  provider: "aws",
  securityReport: {
    "damaged.tf": [
      {
        rule_id: "AVD-AWS-0107",
        long_id: "aws-ec2-no-public-ingress-sgr",
        rule_description:
          "An ingress security group rule allows traffic from /0.",
        rule_provider: "aws",
        rule_service: "ec2",
        impact: "Your port exposed to the internet",
        resolution: "Set a more restrictive cidr range",
        links: [
          "https://aquasecurity.github.io/tfsec/v1.28.1/checks/aws/ec2/no-public-ingress-sgr/",
          "https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group_rule#cidr_blocks",
        ],
        description: "Security group rule allows ingress from public internet.",
        severity: "CRITICAL",
        warning: false,
        status: 0,
        resource: "aws_security_group.no_issue",
        location: {
          filename: "/foo/bar/damaged.tf",
          start_line: 31,
          end_line: 31,
        },
      },
    ],
  },
  documentation: {
    inputs: [
      {
        name: "repositories",
        type: "map(object({force_delete : optional(bool, false)\n    tags : optional(map(any))\n  }))",
        description: "List of repositories to create",
        required: true,
      },
    ],
    modules: [
      {
        name: "vote_service_sg",
        source: "terraform-aws-modules/security-group/aws",
        version: "",
      },
    ],
    outputs: [{ name: "github_roles" }, { name: "registries" }],
    providers: [{ name: "aws" }],
    resources: [
      {
        name: "this",
        type: "ecr_lifecycle_policy",
        source: "hashicorp/aws",
        mode: "managed",
        version: "latest",
      },
    ],
  },
};

describe("<ModuleAnalysisTab /> spec", () => {
  it("renders the ModuleAnalysisTab", () => {
    const view = render(
      <ModuleAnalysisTab
        version={fakeResponse.moduleVersion}
        reports={fakeResponse.securityReport}
        documentation={fakeResponse.documentation}
      />,
    );
    expect(view).toMatchSnapshot();
  });
});
