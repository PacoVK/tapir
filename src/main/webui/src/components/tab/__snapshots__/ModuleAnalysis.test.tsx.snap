// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModuleAnalysisTab /> spec renders the ModuleAnalysisTab 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiTabs-root css-1ujnqem-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            aria-label="module detail tabs"
            class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="0"
              type="button"
            >
              Inputs (1)
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Outputs (2)
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Dependencies (2)
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Resources (1)
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Security Report
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
      <div
        aria-labelledby="module-tab-0"
        id="module-tabpanel-0"
        role="tabpanel"
      >
        <div
          class="MuiBox-root css-16f1609"
        >
          <div
            class="MuiStack-root css-c08hs3-MuiStack-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                <span
                  class="MuiBadge-root css-1c32n2y-MuiBadge-root"
                >
                  <strong>
                    repositories
                  </strong>
                  <span
                    class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary css-rt4zii-MuiBadge-badge"
                  />
                </span>
              </p>
              <div
                class="MuiContainer-root MuiContainer-maxWidthSm css-10a3hyv-MuiContainer-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  List of repositories to create
                </p>
                <pre
                  style="color: rgb(57, 58, 52); font-family: \\"Consolas\\", \\"Bitstream Vera Sans Mono\\", \\"Courier New\\", Courier, monospace; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; font-size: .9em; line-height: 1.2em; tab-size: 4; hyphens: none; padding: 1em; margin: .5em 0px; overflow: auto; border: 1px solid #dddddd; background-color: white;"
                >
                  <code
                    class="language-hcl"
                    style="color: rgb(57, 58, 52); font-family: \\"Consolas\\", \\"Bitstream Vera Sans Mono\\", \\"Courier New\\", Courier, monospace; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; font-size: .9em; line-height: 1.2em; tab-size: 4; hyphens: none;"
                  >
                    <span>
                      map(object({force_delete : optional(bool, false)

                    </span>
                        tags : optional(map(any))

                      })) 
                  </code>
                </pre>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="module-tab-1"
        hidden=""
        id="module-tabpanel-1"
        role="tabpanel"
      />
      <div
        aria-labelledby="module-tab-2"
        hidden=""
        id="module-tabpanel-2"
        role="tabpanel"
      />
      <div
        aria-labelledby="module-tab-3"
        hidden=""
        id="module-tabpanel-3"
        role="tabpanel"
      />
      <div
        aria-labelledby="module-tab-4"
        hidden=""
        id="module-tabpanel-4"
        role="tabpanel"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiTabs-root css-1ujnqem-MuiTabs-root"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
        style="overflow: hidden; margin-bottom: 0px;"
      >
        <div
          aria-label="module detail tabs"
          class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="0"
            type="button"
          >
            Inputs (1)
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Outputs (2)
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Dependencies (2)
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Resources (1)
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Security Report
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      aria-labelledby="module-tab-0"
      id="module-tabpanel-0"
      role="tabpanel"
    >
      <div
        class="MuiBox-root css-16f1609"
      >
        <div
          class="MuiStack-root css-c08hs3-MuiStack-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              <span
                class="MuiBadge-root css-1c32n2y-MuiBadge-root"
              >
                <strong>
                  repositories
                </strong>
                <span
                  class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary css-rt4zii-MuiBadge-badge"
                />
              </span>
            </p>
            <div
              class="MuiContainer-root MuiContainer-maxWidthSm css-10a3hyv-MuiContainer-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                List of repositories to create
              </p>
              <pre
                style="color: rgb(57, 58, 52); font-family: \\"Consolas\\", \\"Bitstream Vera Sans Mono\\", \\"Courier New\\", Courier, monospace; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; font-size: .9em; line-height: 1.2em; tab-size: 4; hyphens: none; padding: 1em; margin: .5em 0px; overflow: auto; border: 1px solid #dddddd; background-color: white;"
              >
                <code
                  class="language-hcl"
                  style="color: rgb(57, 58, 52); font-family: \\"Consolas\\", \\"Bitstream Vera Sans Mono\\", \\"Courier New\\", Courier, monospace; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; font-size: .9em; line-height: 1.2em; tab-size: 4; hyphens: none;"
                >
                  <span>
                    map(object({force_delete : optional(bool, false)

                  </span>
                      tags : optional(map(any))

                    })) 
                </code>
              </pre>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="module-tab-1"
      hidden=""
      id="module-tabpanel-1"
      role="tabpanel"
    />
    <div
      aria-labelledby="module-tab-2"
      hidden=""
      id="module-tabpanel-2"
      role="tabpanel"
    />
    <div
      aria-labelledby="module-tab-3"
      hidden=""
      id="module-tabpanel-3"
      role="tabpanel"
    />
    <div
      aria-labelledby="module-tab-4"
      hidden=""
      id="module-tabpanel-4"
      role="tabpanel"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
